fe/src/main/java/org/apache/doris/qe/SqlModeHelper.java

```diff
@@ -0,0 +1,136 @@
+            (MODE_REAL_AS_FLOAT | MODE_PIPES_AS_CONCAT | MODE_ANSI_QUOTES |
+                    MODE_IGNORE_SPACE | MODE_NOT_USED | MODE_ONLY_FULL_GROUP_BY |
+                    MODE_NO_UNSIGNED_SUBTRACTION | MODE_NO_DIR_IN_CREATE | MODE_ANSI |
+                    MODE_NO_AUTO_VALUE_ON_ZERO | MODE_NO_BACKSLASH_ESCAPES |
+                    MODE_STRICT_TRANS_TABLES | MODE_STRICT_ALL_TABLES | MODE_DEFAULT);
+
+    private final static ImmutableMap<String, Long> sqlModeSet =
+            ImmutableMap.<String, Long>builder()
+                    .put("REAL_AS_FLOAT", MODE_REAL_AS_FLOAT)
+                    .put("PIPES_AS_CONCAT", MODE_PIPES_AS_CONCAT)
+                    .put("ANSI_QUOTES", MODE_ANSI_QUOTES)
+                    .put("IGNORE_SPACE", MODE_IGNORE_SPACE)
+                    .put("NOT_USED", MODE_NOT_USED)
+                    .put("ONLY_FULL_GROUP_BY", MODE_ONLY_FULL_GROUP_BY)
+                    .put("NO_UNSIGNED_SUBTRACTION", MODE_NO_UNSIGNED_SUBTRACTION)
+                    .put("NO_DIR_IN_CREATE", MODE_NO_DIR_IN_CREATE)
+                    .put("ANSI", MODE_ANSI)
+                    .put("NO_AUTO_VALUE_ON_ZERO", MODE_NO_AUTO_VALUE_ON_ZERO)
+                    .put("NO_BACKSLASH_ESCAPES", MODE_NO_BACKSLASH_ESCAPES)
+                    .put("STRICT_TRANS_TABLES", MODE_STRICT_TRANS_TABLES)
+                    .put("STRICT_ALL_TABLES", MODE_STRICT_ALL_TABLES)
+                    .put("DEFAULT", MODE_DEFAULT)
+                    .build();
+
+
+
+
+    public static String parseValue(Long sqlMode) {
+        if (sqlMode == 0 || (sqlMode & ~MODE_ALLOWED_MASK) != 0) {
+            return "";
+        }
+
+        List<String> names = new ArrayList<String>();
+        for (Map.Entry<String, Long> mode : getSupportedSqlMode().entrySet()) {
+            if ((sqlMode & mode.getValue()) != 0) {
+                names.add(mode.getKey());
+            }
+        }
+
+        return Joiner.on(',').join(names);
+    }
+
+    public static Long parseString(String sqlMode) {
+        long value = 0L;
+        List<String> names =
+                Splitter.on(',').trimResults().omitEmptyStrings().splitToList(sqlMode);
+        for (String key : names) {
+            if (getSupportedSqlMode().containsKey(key.toUpperCase())) {
+                value |= getSupportedSqlMode().get(key.toUpperCase());
+            }
+        }
+
+        return value;
+    }
+
+    public static boolean checkValid(String sqlMode) {
+        if (sqlMode == null) {
+            return false;
+        }
+
+        List<String> values =
+                Splitter.on(',').trimResults().omitEmptyStrings().splitToList(sqlMode);
+        for (String key : values) {
+            if (!getSupportedSqlMode().containsKey(key.toUpperCase())) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    public static boolean checkValid(Long sqlMode) {
+        if ((sqlMode & ~MODE_ALLOWED_MASK) != 0) {
```

Comment:
is 0 a valid mode?

Reply:
yes, If 0 as parameter, a empty string is return by this method, which can be return to the client. Besides, a empty string may be read from image, and I think it is better to parse a empty string to 0 that it means nothing had read so that the sql mode is set to nothing.